File: 2022/day1/solution.py
from pathlib import Path

cals = []

batch = []
for l in open(Path(__file__).resolve().parent / 'input.txt'):
  l = l.strip()
  if not l:
    cals.append(sum(map(int, batch)))
    batch = []
  else:
    batch.append(l)

print(max(cals))

print(sum(sorted(cals)[-3:]))

File: 2022/day1/solution.rs
use std::fs;

fn main() {
  let contents = fs::read_to_string("input.txt").expect("fail");

  let mut cals = Vec::<i32>::new();
  for block in contents.split("\n\n") {
    let mut sum = 0;
    for c in block.split("\n") {
      if c.is_empty() {
        continue;
      }
      sum += c.parse::<i32>().unwrap();
    }
    cals.push(sum);
  }

  cals.sort();

  println!("{}", cals.last().expect("please"));

  println!("{}", cals.iter().rev().take(3).sum::<i32>());
}
